File: src/main/java/com/zndevs/toolset/updater/CommandUpdateVersions.java
package com.zndevs.toolset.updater;

/**
 * Stores module versions, a name, and some data
 */

public class CommandUpdateVersions {

    private String commandName = "";
    private String oldVersion = "";
    private String newVersion = "";
    private Object data;

    public CommandUpdateVersions() {
    }

    public CommandUpdateVersions(String commandName, String oldVersion, String newVersion) {
        setCommandName(commandName);
        setOldVersion(oldVersion);
        setNewVersion(newVersion);
    }

    public CommandUpdateVersions(String commandName, String oldVersion, String newVersion, Object otherData) {
        setCommandName(commandName);
        setOldVersion(oldVersion);
        setNewVersion(newVersion);
        setOtherData(otherData);
    }

    public String getCommandName() {
        return commandName;
    }

    void setCommandName(String name) {
        commandName = name;
    }

    public String getOldVersion() {
        return oldVersion;
    }

    void setOldVersion(String version) {
        oldVersion = version;
    }

    public String getNewVersion() {
        return newVersion;
    }

    void setNewVersion(String version) {
        newVersion = version;
    }

    public Object getOtherData() {
        return data;
    }

    void setOtherData(Object data) {
        this.data = data;
    }

}

File: src/main/kotlin/com/zndevs/toolset/updater/CommandUpdateVersions.kt
package com.zndevs.toolset.updater

/**
 * Stores module versions, a name, and some data
 */

class CommandUpdateVersions {

    var commandName = ""
        internal set
    var oldVersion = ""
        internal set
    var newVersion = ""
        internal set
    var otherData: Any? = null
        internal set

    constructor()

    constructor(name: String, oldVer: String, newVer: String) {
        commandName = name
        oldVersion = oldVer
        newVersion = newVer
    }

    constructor(name: String, oldVer: String, newVer: String, otherDta: Any) {
        commandName = name
        oldVersion = oldVer
        newVersion = newVer
        otherData = otherDta
    }

}
